client/src/main/java/no/nav/common/client/axsys/CachedAxsysClient.java

```diff
@@ -0,0 +1,47 @@
+package no.nav.common.client.axsys;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import no.nav.common.health.HealthCheckResult;
+import no.nav.common.types.identer.EnhetId;
+import no.nav.common.types.identer.NavIdent;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import static no.nav.common.client.utils.CacheUtils.tryCacheFirst;
+
+public class CachedAxsysClient implements AxsysClient {
+
+    private final AxsysClient axsysClient;
+
+    private final Cache<NavIdent, AxsysEnheter> hentTilgangerCache = Caffeine.newBuilder()
+            .expireAfterWrite(12, TimeUnit.HOURS)
+            .maximumSize(1000)
+            .build();
+
+
+    private final Cache<EnhetId, List<NavIdent> > hentAnsatteCache = Caffeine.newBuilder()
+            .expireAfterWrite(12, TimeUnit.HOURS)
+            .maximumSize(1000)
+            .build();
+
+    public CachedAxsysClient(AxsysClient axsysClient) {
+        this.axsysClient = axsysClient;
```

Comment:
Tenker det kunne vært greit med en ekstra konstruktør for å spesifisere cacher selv, så kan man overstyre cachen hvis det er ønskelig.